src/lib/hooks/useApproval.ts

```diff
@@ -14,29 +14,41 @@ export enum ApprovalState {
   APPROVED = 'APPROVED',
 }
 
+export interface ApprovalStateForSpenderParams {
+  approvalState: ApprovalState
+}
```

Comment:
What about the return being just `ApprovalState` without the nested object?

Reply:
Also, this is not a fn params, but the hook return

Reply:
> What about the return being just ApprovalState without the nested object?

That's what we had before. I want it to return `currentAllowance` too, but I forgot to add it in they type. 
I will add now.